Assets/Scripts/Enemy.cs

```diff
@@ -2,34 +2,66 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class Enemy : MonoBehaviour
+/*
+ * Abstract class providing framework for enemy script design
+ */
+public abstract class Enemy : MonoBehaviour
 {
+    // Each enemy state has a corresponding behaviour the enemy performs while in that state
+    /* 
+     * TODO: Research if there's a more computationally efficient way to allow editing of EnemyBehaviour members in Inspector than serializing the EnemyBehaviours by reference.
+     * Example possibilities include defining a custom Editor subclass to add EnemyBehaviour members to Inspector
+     * As of 5/6/2023, the project is small enough that the overhead of [SerializeReference] is negligible. This is impermanent and experimentation will be required.
```

Comment:
```suggestion
    /* 
     * TODO: Research if there's a more computationally efficient way to allow editing of EnemyBehaviour members in Inspector than serializing the EnemyBehaviours by reference.
     * Example possibilities include defining a custom Editor subclass to add EnemyBehaviour members to Inspector.
```

Reply:
Interesting. Do we not want to timestamp the most recent time someone looks into this problem?

Reply:
In general I don't see the need. If we decide to revisit this tomorrow or one year from now, it should be dictated by the state of the project rather than an arbitrary timeline. We don't even timestamp on really large projects like scikit-learn.

Reply:
I disagree with the conceptual justification, as I don't find the inclusion of a timestamp to add reliance on an arbitrary timeline, it just adds information that can be evaluated at any later point about the non-arbitrary timeline choice of... the timeline that has happened on the project thus far. It seems like you hold the belief that by timestamping it, it means the same thing in six months (i.e. oh my god we haven't looked at this in six months) regardless of the timeline that actually occurs, hence the emphasis on the arbitrariness of the timeline. But for those of us with moderately good memories of things from six months ago, it provides a reference point to look back to and compare the current state of the project against to evaluate whether the project is now **different** in any key ways that make the TODO more immediately actionable. In essence, it lets us consider not just the current position, but the deltas, which in my opinion does add potentially valuable information to the decision-making process.

That all being said, I legitimately have no preference in this specific case. I don't think the timestamp is going to be particularly necessary for me because my memory is slightly better than moderately good, and I'm often capable of remembering the state of the project in which I last considered an issue like this **without** the timestamp. You clearly have a stronger preference here toward what you consider normal project organization practice, so I'm going to commit the suggestion.